tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+"""Tests for the Hyperion integration."""
+# from tests.async_mock import AsyncMock, MagicMock, patch
+from asynctest import CoroutineMock, Mock, call, patch
+from hyperion import const
+import pytest
+
+from homeassistant.components.hyperion import light as hyperion_light
+from homeassistant.components.light import ATTR_BRIGHTNESS, ATTR_EFFECT, ATTR_HS_COLOR
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+
+TEST_HOST = "test-hyperion-host"
+TEST_NAME = "test-hyperion-name"
+TEST_PRIORITY = 128
+
+TEST_CONFIG = {
+    CONF_HOST: TEST_HOST,
+    CONF_NAME: TEST_NAME,
+    CONF_PORT: const.DEFAULT_PORT,
+    hyperion_light.CONF_PRIORITY: 128,
+}
+
+
+def create_mock_client():
+    """Create a mock Hyperion client."""
+    mock_client = Mock()
+    mock_client.async_connect = CoroutineMock(return_value=True)
+    mock_client.adjustment = None
+    mock_client.effects = None
+    return mock_client
+
+
+def create_hyperion_entity(client=None):
+    """Create a mock Hyperion entity."""
+    entity = hyperion_light.Hyperion(
+        TEST_NAME, TEST_PRIORITY, client or create_mock_client()
+    )
+    return entity
+
+
+def create_client_and_entity():
+    """Create a mock Hyperion client and entity."""
+    client = create_mock_client()
+    return client, create_hyperion_entity(client)
+
+
+async def test_setup_platform(hass):
+    """Test the platform setup."""
+    add_entities = Mock()
+    client = create_mock_client()
+
+    with patch("hyperion.client.HyperionClient", return_value=client):
+        await hyperion_light.async_setup_platform(hass, TEST_CONFIG, add_entities)
```

Comment:
Please set up the integration with `async_setup_component` instead and let the platform add an entity. Then assert the entity state via the core state machine and call the services with the service registry. We should not interact directly with the entity instance at all. This will make the tests more robust.

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/487a74ba5d27e0042f61598d615467503e4202c2/tests/components/manual/test_alarm_control_panel.py#L88-L125